packages/ember-application/lib/system/application-instance.js

```diff
@@ -83,24 +79,20 @@ let ApplicationInstance = EmberObject.extend(RegistryProxy, ContainerProxy, {
   rootElement: null,
 
   init() {
-    this._super(...arguments);
+    let application = get(this, 'application');
```

Comment:
Interacting with `this` before `super()` is not compatible with ES6 classes, fwiw. If this can be differently factored, I think that would be ideal.